ddl/table.go

```diff
@@ -1292,6 +1292,43 @@ func onAlterTablePartitionAttributes(t *meta.Meta, job *model.Job) (ver int64, e
 	return ver, nil
 }
 
+func onAlterTablePartitionOptions(t *meta.Meta, job *model.Job) (ver int64, err error) {
+	var partitionID int64
+	policyRefInfo := &model.PolicyRefInfo{}
+	placementSettings := &model.PlacementSettings{}
+	err = job.DecodeArgs(&partitionID, policyRefInfo, placementSettings)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return 0, errors.Trace(err)
+	}
+	tblInfo, err := getTableInfoAndCancelFaultJob(t, job, job.SchemaID)
+	if err != nil {
+		return 0, err
+	}
+
+	ptInfo := tblInfo.GetPartitionInfo()
+	if ptInfo.GetNameByID(partitionID) == "" {
+		job.State = model.JobStateCancelled
+		return 0, errors.Trace(table.ErrUnknownPartition.GenWithStackByArgs("drop?", tblInfo.Name.O))
+	}
+	for idx, ptDef := range ptInfo.Definitions {
```

Comment:
Looking at `(*PartitionInfo) GetNameByID`, it looks like this could be done similarly, including the check for existence above.
I.e. could you remove the `if ptInfo.GetNameByID(partitionID) ...` and replace the loop here with `for i := range ptInfo.Definitions {` and then check afterwards if the partition was found and if not return the error? Then you will only iterate over the array once, and according to the [note in GetNameByID it is also faster](https://github.com/pingcap/parser/blob/16d022b0dae17fde51c40a84b3d00bce4b037904/model/model.go#L801).

Reply:
done